### PR TITLE
Revert "delete release notes 8.0"

### PR DESCRIPTION
**Pull request by hfxsd**
Reverts pingcap/docs#16891

### CODE REVIEW

File: releases/release-8.0.0.md [hunk 1]
```diff
@@ -5,9 +5,7 @@ summary: Learn about the new features, compatibility changes, improvements, and
 
 # TiDB 8.0.0 Release Notes
 
-> **Note:**
->
-> Currently, TiDB v8.0.0 is still in development. The following content is subject to change before the official release.
+Release date: March 28, 2024
```
Comment (hfxsd) [inline releases/release-8.0.0.md:8]:
```suggestion
Release date: March 29, 2024
```

File: releases/release-notes.md [hunk 2]
```diff
@@ -5,6 +5,10 @@ aliases: ['/docs/dev/releases/release-notes/','/docs/dev/releases/rn/']
 
 # TiDB Release Notes
 
+## 8.0
+
+- [8.0.0-DMR](/releases/release-8.0.0.md): 2024-03-28
```
Comment (hfxsd) [inline releases/release-notes.md:10]:
```suggestion
- [8.0.0-DMR](/releases/release-8.0.0.md): 2024-03-29
```

File: releases/release-timeline.md [hunk 3]
```diff
@@ -9,6 +9,7 @@ This document shows all the released TiDB versions in reverse chronological orde
 
 | Version | Release Date |
 | :--- | :--- |
+| [8.0.0](/releases/release-8.0.0.md) | 2024-03-28 |
```
Comment (hfxsd) [inline releases/release-timeline.md:12]:
```suggestion
| [8.0.0](/releases/release-8.0.0.md) | 2024-03-29 |
```

**Review by lilin90 — APPROVED** [review]
/lgtm

**Pull request merged**